### PR TITLE
Patch/api request retries

### PR DESCRIPTION
**Pull request by jixy2012**
- use tenacity to retry any failed api requests

### CODE REVIEW

**Review by madelonhulsebos — APPROVED** [review]
Nice! General comment: it might be good to version the current version on main for reproducibility of target results!

File: target_benchmark/generators/DefaultGenerator.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 from langchain_openai import ChatOpenAI
+from tenacity import retry, stop_after_attempt, wait_exponential
 
 from target_benchmark.generators.AbsGenerator import AbsGenerator
 from target_benchmark.generators.GeneratorPrompts import (
     DEFAULT_SYSTEM_PROMPT,
     QA_USER_PROMPT,
 )
 
+default_lm = "gpt-4o-mini-2024-07-18"
```
[the remark this answers is not in the document]
Comment (jixy2012) [reply target_benchmark/generators/DefaultGenerator.py]: fixed :)

**Comment by jixy2012** [comment]
> current version on main for

perfect. also just created a release on main to mark the version used for workshop paper evals.

File: target_benchmark/generators/GeneratorPrompts.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## Language Models ##
```
Comment (madelonhulsebos) [inline target_benchmark/generators/GeneratorPrompts.py:1]: Not sure if this is the best place? I'd more expect this setting in a root-level config file.. But we can move it later!

**Review by madelonhulsebos — APPROVED** [review]

File: target_benchmark/generators/GeneratorPrompts.py (hunk 2 above)
Comment (jixy2012) [reply target_benchmark/generators/GeneratorPrompts.py:1]: added an [issue](https://github.com/target-benchmark/target/issues/28#issue-2616774714) for later

**Pull request merged**